### PR TITLE
Moved the creation of the searchResult to the searchQueue item.

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read [contribution guide](https://github.com/PyMedusa/SickRage/blob/master/.github/CONTRIBUTING.md)

* Changed searchResult to snakecase.
* Removed unneeded params for snatchSelection() method.
* Changed mako to not send unneeded params. rowid is enough.
* Added check for getting the show_obj, chances are slim, but maybe a user leaves a snatchSelection window open, while deleting the show.
* Some pep8 changes.
* Added docstrings to webserve.py and search_queue.py methods.

### CODE REVIEW

File: sickbeard/search_queue.py [hunk 1]
```diff
@@ -234,32 +234,57 @@ def run(self):
...
 
-            if self.searchResult:
-                if self.searchResult.seeders not in (-1, None) and self.searchResult.leechers not in (-1, None):
+            if search_result:
+                if search_result.seeders not in (-1, None) and search_result.leechers not in (-1, None):
                     logger.log(u"Downloading {0} with {1} seeders and {2} leechers from {3}".
-                               format(self.searchResult.name,
-                                      self.searchResult.seeders, self.searchResult.seeders, self.searchResult.provider.name))
+                               format(search_result.name,
+                                      search_result.seeders, search_result.seeders, search_result.provider.name))
```
Comment (medariox) [inline sickbeard/search_queue.py]: Second search_result.seeders should be search_result.leechers.

File: sickbeard/webserve.py [hunk 2]
```diff
@@ -1401,55 +1401,61 @@ def titler(x):
...
         try:
-            show = int(show)  # fails if show id ends in a period SickRage/sickrage-issues#65
+            show = int(cached_result['indexerid'])  # fails if show id ends in a period SickRage/sickrage-issues#65
             show_obj = Show.find(sickbeard.showList, show)
         except (ValueError, TypeError):
-            return self._genericMessage("Error", "Invalid show ID: {}".format(show))
+            return self._genericMessage("Error", "Invalid show ID: {0}".format(show))
+
+        if not show_obj:
+            return self._genericMessage("Error", "Could not find a show with id {0} in the list of shows, did you remove the show? {}".format(show))
```
Comment (medariox) [inline sickbeard/webserve.py]: Last {} should be removed.

File: sickbeard/webserve.py [hunk 3]
```diff
@@ -1401,55 +1401,61 @@ def titler(x):
...
-        sickbeard.searchQueueScheduler.action.add_item(ep_queue_item)
+        # Create the queue item
+        snatch_queue_item = search_queue.ManualSnatchQueueItem(show_obj, ep_objs, provider, cached_result)
+
+        # Add the queue item to the queue
+        sickbeard.searchQueueScheduler.action.add_item(snatch_queue_item)
 
-        while ep_queue_item.success is not False:
-            if ep_queue_item.started and ep_queue_item.success:
+        while snatch_queue_item.success is not False:
```
Comment (medariox) [inline sickbeard/webserve.py]: I would use ```while snatch_queue_item.success:``` instead.

Comment (p0psicles) [inline sickbeard/webserve.py]: Then it will skip it first try. As the value starts with None.

File: sickbeard/webserve.py (hunk 2 above)
Comment (p0psicles) [inline sickbeard/webserve.py]: a fck missed that

**Comment by p0psicles** [comment]
Fixed from @medariox comments, and force pushed.

**Comment by medariox** [comment]
Approved

**Pull request merged**